sql: show ttl_job_cron if ttl_job_cron is not explicitly set

otan opened the pull request:
Resolves #80215

Release note (sql change): ttl_job_cron is now displayed on SHOW CREATE
TABLE and `reloptions` by default.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80221)
<!-- Reviewable:end -->

rafiss reviewed (approved):
nice!

> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80221#-:-N07-KM87H7RNla7Yv1Y:b-7m74ox)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

rafiss commented:
simple test fix needed:
```
=== RUN   TestDataDriven/row_level_ttl
    datadriven_test.go:304:
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/testdata/backup-restore/row_level_ttl:35: SELECT create_statement FROM [SHOW CREATE TABLE d.public.t]
        output didn't match expected:
        @@ -1,6 +1,6 @@
         CREATE TABLE public.t (
           id INT8 NOT NULL,
           crdb_internal_expiration TIMESTAMPTZ NOT VISIBLE NOT NULL DEFAULT current_timestamp():::TIMESTAMPTZ + '00:10:00':::INTERVAL ON UPDATE current_timestamp():::TIMESTAMPTZ + '00:10:00':::INTERVAL,
           CONSTRAINT t_pkey PRIMARY KEY (id ASC)
        -) WITH (ttl = 'on', ttl_automatic_column = 'on', ttl_expire_after = '00:10:00':::INTERVAL)
        +) WITH (ttl = 'on', ttl_automatic_column = 'on', ttl_expire_after = '00:10:00':::INTERVAL, ttl_job_cron = '@hourly')
```

otan committed:
sql: show ttl_job_cron if ttl_job_cron is not explicitly set

Release note (sql change): ttl_job_cron is now displayed on SHOW CREATE
TABLE and `reloptions` by default.

otan commented:
bors r=rafiss
